### DISCUSSION TITLE
provide an option to show all keys pressed

### DISCUSSION THREAD

**Discussion by jerabaul29** [helix-editor/helix#7044] [General]
It is quite convenient to be able to record videos / share screen while showing all keys pressed for tutorial purposes. For example (I think this is done in vim), see in the bottom right corner: https://user-images.githubusercontent.com/15617291/128964224-f157022c-25cd-4a60-a0da-7d1462564ae4.gif .

Would it be possible to provide an option and command to turn it on / off that displays all keys pressed? I tried to find something similar in the ```esc - ?``` command explorer without success, but maybe I am missing something :) .

**Comment by David-Else** [comment]
The software that is showing the keys pressed in that video is [screenkey](https://www.thregr.org/~wavexx/software/screenkey/) and is great. Providing you use the best operating system, Linux, you can start using it today!

VS Code has a key showing feature built in. The advantage of having it built into Helix would be that it could auto turn off in insert mode, that could be a cool feature. You can work around it in screenkey by turning it on and off on the fly holding both ctrl keys at once. It would be easier to get screen key and Helix communicating somehow, I am sure it can be done...

NOTE: It only works when using Xorg, no Wayland. There is another newer less fully featured program for that, [show me the key](https://github.com/AlynxZhou/showmethekey)

**Comment by mudokenking** [comment]
wshowkeys for wayland
